scripts/variantstore/wdl/GvsCreateVATFromAnnotations.wdl

```diff
@@ -0,0 +1,549 @@
+          --annotated_json ~{updated_annotation_json} \
+          --output_vt_json ~{output_vt_json} \
+          --output_genes_json ~{output_genes_json}
+
+        gsutil cp ~{output_vt_json} '~{output_vt_gcp_path}'
+        gsutil cp ~{output_genes_json} '~{output_genes_gcp_path}'
+
+     >>>
+    # ------------------------------------------------
+    # Runtime settings:
+    runtime {
+        docker: "us.gcr.io/broad-dsde-methods/variantstore:rc_vat_update_2022_05_06"
+        memory: "8 GB"
+        preemptible: 5
+        cpu: "1"
+        disks: "local-disk 250 SSD"
+    }
+    # ------------------------------------------------
+    # Outputs:
+    output {
+        File vat_vt_json="~{output_vt_json}"
+        File vat_genes_json="~{output_genes_json}"
+        Boolean done = true
+    }
+}
+
+task BigQueryLoadJson {
+    meta { # since the WDL will not see the updated data (its getting put in a gcp bucket)
+        volatile: true
+    }
+
+    input {
+        File? nirvana_schema
+        File? vt_schema
+        File? genes_schema
+        String project_id
+        String dataset_name
+        String output_path
+        String table_suffix
+        String? service_account_json_path
+        Array[String] prep_jsons_done
+    }
+
+    # There are two pre-vat tables. A variant table and a genes table. They are joined together for the vat table
+
+    String vat_table = "vat_" + table_suffix
+    String variant_transcript_table = "vat_vt_"  + table_suffix
+    String genes_table = "vat_genes_" + table_suffix
+
+    String vt_path = output_path + 'vt/*'
+    String genes_path = output_path + 'genes/*'
+
+    String has_service_account_file = if (defined(service_account_json_path)) then 'true' else 'false'
+
+    command <<<
+
+       echo "project_id = ~{project_id}" > ~/.bigqueryrc
+
+       DATE=86400 ## 24 hours in seconds
+
+
+       if [ ~{has_service_account_file} = 'true' ]; then
+            gsutil cp ~{service_account_json_path} local.service_account.json
+            export GOOGLE_APPLICATION_CREDENTIALS=local.service_account.json
+            gcloud auth activate-service-account --key-file=local.service_account.json
+            gcloud config set project ~{project_id}
+       fi
+
+       set +e
+       bq show --project_id ~{project_id} ~{dataset_name}.~{variant_transcript_table} > /dev/null
+       BQ_SHOW_RC=$?
+       set -e
+
+       if [ $BQ_SHOW_RC -ne 0 ]; then
+         echo "Creating a pre-vat table ~{dataset_name}.~{variant_transcript_table}"
+         bq --location=US mk --expiration=$DATE --project_id=~{project_id}  ~{dataset_name}.~{variant_transcript_table} ~{vt_schema}
+       fi
+
+       echo "Loading data into a pre-vat table ~{dataset_name}.~{variant_transcript_table}"
+       echo ~{vt_path}
+       echo ~{genes_path}
+       bq --location=US load --project_id=~{project_id} --source_format=NEWLINE_DELIMITED_JSON ~{dataset_name}.~{variant_transcript_table} ~{vt_path}
```

Comment:
Is it okay to proceed with loading data into the variant transcript table if it already existed? The seemingly similar logic for the vat table on line 248 `rm`s the table if it already existed.

Reply:
yes---the logic was originally created such that we could keep adding data to the two temp tables--the Variant-Transcript table and the Genes table--and then a fresh VAT would be created based on that.

Now that the usage has changed a bit, reworking the logic and failsafes here would all be helpful